flickit-assessment-data/src/main/java/org/flickit/assessment/data/jpa/kit/kitversion/KitVersionJpaRepository.java

```diff
@@ -17,8 +17,11 @@ public interface KitVersionJpaRepository extends JpaRepository<KitVersionJpaEnti
     @Modifying
     @Query("""
             UPDATE KitVersionJpaEntity k
-            SET k.status = :status
+            SET k.status = :status,
+                k.statusVersion = :statusVersion
             WHERE k.id = :id
         """)
-    void updateStatus(@Param("id") long kitVersionId, @Param("status") int status);
+    void updateStatus(@Param("id") long kitVersionId,
+                      @Param("status") int status,
+                      @Param("statusVersion") long statusVersion);
```

Comment:
**Suggestion:** Ensure that the updateStatus method in the repository handles potential null values or exceptions when the statusVersion is calculated based on the status and id, especially when the status is not 0, 1, or 2. [possible issue, importance: 6]
```suggestion
    void updateStatus(@Param("id") long kitVersionId,
                      @Param("status") int status,
                      @Param("statusVersion") Long statusVersion);
```